File: update_favicon.js
const fs = require('fs');

fs.readdirSync('.').forEach(file => {
    if (file.endsWith('.html')) {
        let content = fs.readFileSync(file, 'utf-8');
        let newContent = content.replace(/href="assets\/favicon\.png"/g, 'href="assets/g-favicon.png"');
        if (content !== newContent) {
            fs.writeFileSync(file, newContent, 'utf-8');
            console.log('Updated ' + file);
        }
    }
});

File: update_favicon.py
import os
import glob

html_files = glob.glob('*.html')
for file_path in html_files:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Replace all favicon references
    new_content = content.replace('href="assets/favicon.png"', 'href="assets/g-favicon.png"')
    
    if new_content != content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f'Updated {file_path}')
